functions/Get-DbaDbFileMapping.ps1

```diff
@@ -0,0 +1,90 @@
+function Get-DbaDbFileMapping {
+    <#
+    .SYNOPSIS
+        Provides an easy way to build the FileMapping Hashtable for Restore-DbaDatabase from an existing database
+
+    .DESCRIPTION
+        Provides an easy way to build the FileMapping Hashtable for Restore-DbaDatabase from an existing database
+
+    .PARAMETER SqlInstance
+        The target SQL Server instance or instances. This can be a collection and receive pipeline input.
+
+    .PARAMETER SqlCredential
+        Login to the target instance using alternative credentials. Accepts PowerShell credentials (Get-Credential).
+
+        Windows Authentication, SQL Server Authentication, Active Directory - Password, and Active Directory - Integrated are all supported.
+
+        For MFA support, please use Connect-DbaInstance.
+
+    .PARAMETER Database
+        The database(s) to process - this list is auto-populated from the server. If unspecified, all databases will be processed.
+
+   .PARAMETER InputObject
+        Database object piped in from Get-DbaDatabase
+
+    .PARAMETER EnableException
+        By default, when something goes wrong we try to catch it, interpret it and give you a friendly warning message.
+        This avoids overwhelming you with "sea of red" exceptions, but is inconvenient because it basically disables advanced scripting.
+        Using this switch turns this "nice by default" feature off and enables you to catch exceptions with your own try/catch.
+
+    .NOTES
+        Tags: Database
+        Author: Chrissy LeMaire (@cl), netnerds.net | Andreas Jordan (@JordanOrdix), ordix.de
+
+        Website: https://dbatools.io
+        Copyright: (c) 2021 by dbatools, licensed under MIT
+        License: MIT https://opensource.org/licenses/MIT
+
+    .LINK
+        https://dbatools.io/Get-DbaDbFileMapping
+
+    .EXAMPLE
+        PS C:\> $filemap = Get-DbaDbFileMapping -SqlInstance sql2016 -Database test
+        PS C:\> Get-ChildItem \\nas\db\backups\test | Restore-DbaDatabase -SqlInstance sql2019 -Database test -FileMapping $filemap.FileMapping
+
+        Restores test to sql2019 using the file structure built from the existing database on sql2016
```

Comment:
I would add this example to the restore command as well and then I'm good to merge

Reply:
Might replace the example I think the restore command has for manually building the hashtable object.

Reply:
I'll add it in addition to -- I think it's important to show ppl how to build the hashtable.

Reply:
Ty for the approval! Will merge once tests finish.